src/OpenApi/src/Services/OpenApiOptions.cs

```diff
@@ -34,4 +38,55 @@ public OpenApiOptions()
     /// A delegate to determine whether a given <see cref="ApiDescription"/> should be included in the given OpenAPI document.
     /// </summary>
     public Func<ApiDescription, bool> ShouldInclude { get; set; }
+
+    /// <summary>
+    /// Registers a new document transformer on the current <see cref="OpenApiOptions"/> instance.
+    /// </summary>
+    /// <typeparam name="TTransformerType">The type of the <see cref="IOpenApiDocumentTransformer"/> to instantiate.</typeparam>
+    /// <returns>The <see cref="OpenApiOptions"/> instance for further customization.</returns>
+    public OpenApiOptions UseTransformer<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TTransformerType>()
+        where TTransformerType : IOpenApiDocumentTransformer
+    {
+        DocumentTransformers.Add(new ActivatedOpenApiDocumentTransformer(typeof(TTransformerType)));
```

Comment:
Does this need to be thread-safe?  I'm assuming all configuration happens on a single thread, so it doesn't?

Reply:
> Does this need to be thread-safe? 

Good question! AFAIK, we don't use thread-safe collection types in other Options objects (`AuthenticationOptions`, RouteOptions`, etc.) so I think we're safe here.